### PR TITLE
8296610: java/net/HttpURLConnection/SetAuthenticator/HTTPSetAuthenticatorTest.java failed with "BindException: Address already in use: connect"

### PR DESCRIPTION
**Pull request by dfuch**
The test has been observed failing once in the CI, on windows, with a BindException (address already in use) on the client side. This usually means that the system has run out of ephemeral ports or file descriptors. The proposed fix will catch the BindException, run the gc, wait a bit, and retry once. No guarantee that this will actually work - but maybe worth a try, though I worry a bit the authentication count might get wrong if the request that gets the BindException is the first one that follows the 401/407 response - in case the connection was closed after receiving 401/407. Since we have no visibility of which call to connect() fails, it is hard to cater for this possibility without making the test too permissive.

The alternative would be to throw a SkipException in case BindException is received.

While working on the fix I noticed that some parts of the test had been commented out waiting for a RFE to be implemented. Since the RFE has been implemented since then, I went out and uncommented that part of the test.

Comments welcome.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296610](https://bugs.openjdk.org/browse/JDK-8296610): java/net/HttpURLConnection/SetAuthenticator/HTTPSetAuthenticatorTest.java failed with "BindException: Address already in use: connect"

### Reviewers
 * [Michael McMahon](https://openjdk.org/census#michaelm) (@Michael-Mc-Mahon - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11634/head:pull/11634` \
`$ git checkout pull/11634`

Update a local copy of the PR: \
`$ git checkout pull/11634` \
`$ git pull https://git.openjdk.org/jdk pull/11634/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11634`

View PR using the GUI difftool: \
`$ git pr show -t 11634`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11634.diff">https://git.openjdk.org/jdk/pull/11634.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/net/HttpURLConnection/SetAuthenticator/HTTPTestClient.java [hunk 1]
```diff
@@ -40,6 +41,28 @@ public static void connect(HttpProtocolType protocol,
...
+        try {
+            doConnect(protocol, server, authType, auth);
+        } catch (BindException ex) {
+            // sleep a bit then try again once
+            System.out.println("WARNING: Unexpected BindException: " + ex);
+            System.out.println("\tSleeping a bit and try again...");
+            System.gc();
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException iex) {
```
Comment (Michael-Mc-Mahon) [inline test/jdk/java/net/HttpURLConnection/SetAuthenticator/HTTPTestClient.java:53]: If it only fails rarely, would it be safer to delay a few seconds to allow more time for resources to be freed?

Comment (dfuch) [reply test/jdk/java/net/HttpURLConnection/SetAuthenticator/HTTPTestClient.java:53]: Good point. What delay would you recommend?

Comment (Michael-Mc-Mahon) [reply test/jdk/java/net/HttpURLConnection/SetAuthenticator/HTTPTestClient.java:53]: 2-3 sec maybe

**Review by Michael-Mc-Mahon — APPROVED** [review]
LGTM

**Pull request closed without merging**